misc/python/materialize/checks/range.py

```diff
@@ -0,0 +1,110 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+from textwrap import dedent
+from typing import List
+
+from materialize.checks.actions import Testdrive
+from materialize.checks.checks import Check
+
+
+class Range(Check):
```

Comment:
As discussed with Philip, create a source using `CREATE SOURCE generator1 FROM LOAD GENERATOR COUNTER WITH (SIZE = '1')` and use `generator1_progress` in materialized views.

Reply:
I am sorry, the source needs to be of type KAFKA for the _progress table to have a RANGE column. You can check the other .py files in the directory for an example of such source . It can be `ENVELOPE NONE`, so there is no need for schemas and such.